### PR TITLE
Make ajax validation unique for each plugin

### PR DESCRIPTION
**Pull request by MlKilderkin**
## What

Address #22 
Make ajax validation unique for each plugin

### CODE REVIEW

**Review by jamesckemp — CHANGES_REQUESTED** [review]
@MlKilderkin Did you have to make any changes to Flux/Account Pages when testing this? We don't use the `do_settings_field()` method as we already have our own settings. We're directly redering the license field, which means the new plugin slug is not set:

```
/**
 * Output license field.
 */
public static function license_field() {
	$container = Config::get_container();
	$container->get( License_Field::class )->render( false, false );
}
```

**Comment by MlKilderkin** [comment]
> @MlKilderkin Did you have to make any changes to Flux/Account Pages when testing this? We don't use the `do_settings_field()` method as we already have our own settings. We're directly redering the license field, which means the new plugin slug is not set:
> 
> ```
> /**
>  * Output license field.
>  */
> public static function license_field() {
> 	$container = Config::get_container();
> 	$container->get( License_Field::class )->render( false, false );
> }
> ```
@jamesckemp even in that case render method with call `Field.php` class and `do_settings_fields`. It is a custom method not a WP default

**Comment by MlKilderkin** [comment]
And I haven't touch anything in plugin. Only Uplink library has been updated

**Comment by MlKilderkin** [comment]
@jamesckemp Could you check that you have `window` variable with required items? 
For example:
```
stellarwp_config_iconic_wap: {action: 'pue-validate-key-uplink-iconic_wap'}
stellarwp_config_sampletest: {action: 'pue-validate-key-uplink-sampletest'}
```

File: src/Uplink/Admin/License_Field.php [hunk 1]
```diff
@@ -91,14 +91,15 @@ public function render( bool $show_title = true, bool $show_button = true ) {
...
 		$js_src = apply_filters( 'stellarwp/uplink/' . Config::get_hook_prefix() . '/admin_js_source', $path . '/assets/js/key-admin.js' );
-
 		wp_register_script( $handle, $js_src, [ 'jquery' ], '1.0.0', true );
 		wp_enqueue_script( $handle );
+		$action_postfix = strtolower( str_replace( '-', '_', sanitize_title( Config::get_hook_prefix() ) ) );
+		wp_localize_script( $handle, sprintf( 'stellarwp_config_%s', $action_postfix ), [ 'action' => sprintf( 'pue-validate-key-uplink-%s', $action_postfix ) ] );
 
 		$css_src = apply_filters( 'stellarwp/uplink/' . Config::get_hook_prefix() . '/admin_css_source', $path . '/assets/css/main.css' );
-		wp_enqueue_style( 'stellarwp-uplink-license-admin', $css_src );
+		wp_enqueue_style( sprintf( 'stellarwp-uplink-license-admin-%s', Config::get_hook_prefix() ), $css_src );
```
Comment (jamesckemp) [inline src/Uplink/Admin/License_Field.php:103]: Could this use `$handle` instead?

File: src/Uplink/Admin/License_Field.php [hunk 2]
```diff
@@ -91,14 +91,15 @@ public function render( bool $show_title = true, bool $show_button = true ) {
...
 	 */
 	public function enqueue_assets() {
-		$handle = 'stellarwp-uplink-license-admin';
+		$handle = sprintf( 'stellarwp-uplink-license-admin-%s', Config::get_hook_prefix() );
 		$path   = preg_replace( '/.*\/vendor/', plugin_dir_url( $this->get_plugin()->get_path() ) . 'vendor', dirname( __DIR__, 2 ) );
 		$js_src = apply_filters( 'stellarwp/uplink/' . Config::get_hook_prefix() . '/admin_js_source', $path . '/assets/js/key-admin.js' );
-
 		wp_register_script( $handle, $js_src, [ 'jquery' ], '1.0.0', true );
 		wp_enqueue_script( $handle );
+		$action_postfix = strtolower( str_replace( '-', '_', sanitize_title( Config::get_hook_prefix() ) ) );
```
Comment (jamesckemp) [inline src/Uplink/Admin/License_Field.php:99]: We process the slug like this in a few places, should we instead add a config method? `Config::get_hook_prefix_underscored()` or similar?

Comment (MlKilderkin) [reply src/Uplink/Admin/License_Field.php]: Done

File: src/Uplink/Admin/Field.php [hunk 3]
```diff
@@ -88,7 +88,8 @@ public function field_html( array $args = [] ) {
 			esc_attr( $args['html_classes'] ?: '' ),
 			$this->get_html_content( $args ),
 			$this->add_nonce_field(),
-			$args['plugin']
+			$args['plugin'],
+			strtolower( str_replace( '-', '_', sanitize_title( Config::get_hook_prefix() ) ) )
```
Comment (jamesckemp) [inline src/Uplink/Admin/Field.php:92]: Just this one needs replacing!

**Review by jamesckemp — CHANGES_REQUESTED** [review]

**Review by jamesckemp — APPROVED** [review]
Working well 🙌

**Pull request merged**